### DISCUSSION TITLE
Fix images for Ubuntu and Debian @ CM3?

### DISCUSSION THREAD

**Discussion by black-device** [radxa-repo/rbuild#47] [General]
Hello,
It seems that since Builds 24 and 25, (which are apparently the last ones that work on CM3 IO at least) the following ones are having problems, most of them do not even reach login.

Can we know what state the development is in to be able to generate stable images of Ubuntu and Debian both in the CLI and in some desktop for CM3?

Thank you!

**Comment by RadxaYuntian** [comment]
CLI currently has a known issue with the GPU driver. Right now the only supported system variant is the Debian Desktop so I'd suggest you using that one until we resolve the issue with the GPU.

**Reply by black-device** [reply]
Debian desktop works, yes.

Ok thanks for informing.
Do you have any idea when the issue is expected to be fixed for CLI builds?

**Reply by RadxaYuntian** [reply]
The current issue is that the Mesa shipped in Debian 11 does not support GPU used in RK356X. So even when we fixed the boot lock up, the display was still broken, and a lot of users do want to use display even when using CLI image.

As such the current plan is to fix CLI image with Rockchip Debian 12 SDK. Currently there is no timeline for this.

**Reply by black-device** [reply]
I have seen that there are new CLI releases for CM3, but from what I have been able to test, they still have the same problem, right?

**Reply by RadxaYuntian** [reply]
System released after [kernel release 5.10.160-27](https://github.com/radxa-pkg/linux-rk356x/releases/tag/5.10.160-27) should have this fixed.

**Reply by black-device** [reply]
OK, but
sudo ./rbuild --compress --native-build --shrink radxa-cm3-io jammy cli

Still trying to add kernel 5.10.160.26 instead of 27, which is already latest.

Could you know how to make rbuild point to the latest available package?

**Reply by RadxaYuntian** [reply]
First off, we do not officially support Ubuntu.

You will need to use -T option (mentioned in `rbuild`'s help text) to use the test repo in order to use this kernel. It is not released to the production channel yet.

**Reply by black-device** [reply]
okay! thanks for your help.
Anyway, it seems that something is wrong, since I have tried both radxa-cm3-io and radxa-cm3-rpi-cm4-io and neither of the two system images manage to boot, or at least nothing comes out through the hdmi of the evaluation boards (radxa CM3IO, Raspberry CM4IO) with a CM3 rk3566.
Have you tested if the images work for you?
I'll try Debian now, but the Ubuntu ones don't work for me for now.
